Sparse - SpMV: removing cusparse < 11500 path

lucbv opened the pull request:
With the new compiler requirements the oldest supported version of cusparse is that 11500 that comes with cuda-clang 15.0.0

lucbv committed:
Sparse - SpMV: removing cusparse < 11500 path

With the new compiler requirements the oldest supported version of
cusparse is that 11500 that comes with cuda-clang 15.0.0

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

ndellingwood commented on sparse/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp:
```diff
@@ -78,19 +74,11 @@ void spmv_cusparse(const Kokkos::Cuda& exec, Handle* handle, const char mode[],
-#endif
 
   // In CUDA 11.2.0, the algorithm enums were renamed.
   // This corresponds to CUSPARSE_VERSION >= 11400.
```
```suggestion
```

Old comment can be deleted as well, not blocking

ndellingwood commented on sparse/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp:
```diff
@@ -78,19 +74,11 @@ void spmv_cusparse(const Kokkos::Cuda& exec, Handle* handle, const char mode[],
   // matrices. After 11.2.1, the default is performant for imbalanced matrices,
   // and ALG2 now means something else. CUDA >= 11.2.1 corresponds to
   // CUSPARSE_VERSION >= 11402.
```
```suggestion
After CUDA 11.2.1, the default is performant for imbalanced matrices
```

Old comment can be deleted as well, though the info might be a helpful reference reminder? not blocking

ndellingwood reviewed (approved):
nice clean up @lucbv 👍

lucbv committed:
Sparse - spmv: fixing code removal for cusparse < 11500

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

lucbv committed:
Applying clang-format

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

brian-kelley commented:
@lucbv Lines 123-130 of KokkosSparse_spmv_handle.hpp and the version macro logic around it can go away too, since that's where CuSparse9_SpMV_Data is defined.

brian-kelley commented:
Oh never mind, that still gets used in KokkosSparse_spmv_bsrmatrix_tpl_spec_decl.hpp (assuming that is intended to be another PR)

lucbv committed:
Sparse: removing older cusparse code paths

With Kokkos 5.0 we can now assume that we have at least
cusparse 11.5 or 11500 available.

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

lucbv commented:
Trying to remove a few more things in bsr and other places like perf-tests and utils. Hopefully that does not break anything.
Can't really test the cuda-clang at the moment, could be something that we set up on Kumquat.
